client/rpc-servers/src/lib.rs

```diff
@@ -91,24 +91,23 @@ mod inner {
 						return;
 					}
 				};
-
+				// TODO: (dp) DRY this up; it's the same as the WS code
 				let handle = server.stop_handle();
-
-				server.register_module(module).expect("infallible already checked; qed");
 				let mut methods_api = RpcModule::new(());
-				let mut methods = server.method_names();
-				methods.sort();
+				let mut available_methods = module.method_names().collect::<Vec<_>>();
+				available_methods.sort();
```

Comment:
Something that just popped in my head: there are no duplicate names here, and even if there were we wouldn't care about preserving their ordering, so unstable sort is faster.

```suggestion
				available_methods.sort_unstable();
```